citesphere/src/main/java/edu/asu/diging/citesphere/web/user/authorities/AuthorityEntryController.java

```diff
@@ -11,31 +12,95 @@
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.core.Authentication;
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.SessionAttribute;
+import org.springframework.web.bind.annotation.SessionAttributes;
+import org.springframework.web.context.annotation.SessionScope;
 
+import edu.asu.diging.citesphere.core.exceptions.AuthorityImporterNotFoundException;
 import edu.asu.diging.citesphere.core.exceptions.AuthorityServiceConnectionException;
 import edu.asu.diging.citesphere.core.exceptions.GroupDoesNotExistException;
 import edu.asu.diging.citesphere.core.service.IAuthorityService;
 import edu.asu.diging.citesphere.model.authority.IAuthorityEntry;
+import edu.asu.diging.citesphere.model.authority.impl.AuthorityEntry;
+import edu.asu.diging.citesphere.web.user.AuthoritySearchResult;
 import edu.asu.diging.citesphere.user.IUser;
 import edu.asu.diging.citesphere.web.user.FoundAuthorities;
 
 @Controller
+@SessionScope
 public class AuthorityEntryController {
-    
+
     private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private AuthoritySearchResult authorityConceptpowerResult;
     
     @Autowired
     private IAuthorityService authorityService;
+    
+    @RequestMapping("/auth/authority/{zoteroGroupId}/find/userAuthorities")
+    public ResponseEntity<AuthoritySearchResult> getUserAuthorities(Model model, Authentication authentication,
+            @PathVariable("zoteroGroupId") String zoteroGroupId,
+            @RequestParam(defaultValue = "0", required = false, value = "page") int page,
+            @RequestParam(defaultValue = "10", required = false, value = "pageSize") int pageSize,
+            @RequestParam("firstName") String firstName, @RequestParam("lastName") String lastName) {
+
+        AuthoritySearchResult authorityResult = new AuthoritySearchResult();
+
+        List<IAuthorityEntry> userEntries = authorityService.findByName((IUser) authentication.getPrincipal(),
+                firstName, lastName, page, pageSize);
+
+        authorityResult.setFoundAuthorities(userEntries);
+        authorityResult.setCurrentPage(page + 1);
+        authorityResult.setTotalPages(authorityService
+                .getTotalUserAuthoritiesPages((IUser) authentication.getPrincipal(), firstName, lastName, pageSize));
+        return new ResponseEntity<AuthoritySearchResult>(authorityResult, HttpStatus.OK);
+    }
+    
+    @RequestMapping("/auth/authority/{zoteroGroupId}/find/searchAuthorities/{source}")
```

Comment:
and this `.../find/authorities/{source}` ("search" is duplicating "find" here)

Reply:
I'm not sure if Spring will have an issue with two requestmapping matching a url (`.../find/authorities/user` would match both request mappings). If it does, you might have to add a regex expression here in the mapping to exclude "user".